policybot/pkg/testflakes/testflakes.go

```diff
@@ -0,0 +1,204 @@
+					passMap = []*store.TestResult{}
+				}
+				passMap = append(passMap, result)
+				shaMap[testPassed] = passMap
+			} else {
+				shaMap = map[bool][]*store.TestResult{}
+				passMap := []*store.TestResult{}
+				passMap = append(passMap, result)
+				shaMap[testPassed] = passMap
+			}
+			testMap[sha] = shaMap
+		} else {
+			shaMap := map[bool][]*store.TestResult{}
+			results := []*store.TestResult{}
+			results = append(results, result)
+			shaMap[testPassed] = results
+			testMap = map[string]map[bool][]*store.TestResult{}
+			testMap[sha] = shaMap
+		}
+		resultMap[testName] = testMap
+	}
+	return resultMap
+}
+
+// Process the map returned from ProcessResults to check if one test has multiple TestPass values
+// coexisting at the same time. If for one Pull Request the test passed and failed for different runs
+// we mark the test to be flaky.
+func (f *FlakeTester) CheckResults(resultMap map[string]map[string]map[bool][]*store.TestResult) []*FlakyResult {
+	flakyResults := []*FlakyResult{}
+	for testName, testMap := range resultMap {
+		for _, shaMap := range testMap {
+			flakyResult := &FlakyResult{}
+			flakyResult.TestName = testName
+			flakyResult.IsFlaky = false
+			if len(shaMap) > 1 {
+				flakyResult.IsFlaky = true
+			}
+			failFirst := store.TestResult{
+				OrgLogin: "",
+				RepoName: "",
+			}
+			if shaMap[false] != nil {
+				failedTests := shaMap[false]
+				failFirst = *failedTests[0]
+				flakyResult.OrgLogin = failFirst.OrgLogin
+				flakyResult.RepoName = failFirst.RepoName
+				for _, fail := range failedTests {
+					if flakyResult.PrNum == 0 {
+						flakyResult.PrNum = fail.PullRequestNumber
+					}
+					if strings.Compare(flakyResult.LastFail, "") != 0 {
+						if flakyResult.failResult.FinishTime.Before(fail.FinishTime) {
+							flakyResult.failResult = fail
+							flakyResult.LastFail = fail.RunPath
+						}
+					} else {
+						flakyResult.failResult = fail
+						flakyResult.LastFail = fail.RunPath
+					}
+				}
+			}
+			if shaMap[true] != nil {
+				passedTests := shaMap[true]
+				if strings.Compare(failFirst.OrgLogin, "") == 0 {
+					passFirst := passedTests[0]
+					flakyResult.RepoName = passFirst.RepoName
+					flakyResult.OrgLogin = passFirst.OrgLogin
+				}
+				for _, pass := range passedTests {
+					if flakyResult.PrNum == 0 {
+						flakyResult.PrNum = pass.PullRequestNumber
+					}
+					if strings.Compare(flakyResult.LastPass, "") != 0 {
+						if flakyResult.passResult.FinishTime.Before(pass.FinishTime) {
+							flakyResult.LastPass = pass.RunPath
+							flakyResult.passResult = pass
+						}
+					} else {
+						flakyResult.passResult = pass
+						flakyResult.LastPass = pass.RunPath
+					}
+				}
+			}
+			flakyResults = append(flakyResults, flakyResult)
+		}
+	}
+	return flakyResults
+}
+
+// Chase function add issue comment about the flake results
+func (f *FlakeTester) Chase(context context.Context, flakeResults []*FlakyResult, message string) {
```

Comment:
Is "Chase" a meaningful name in this context? Seems like cut&paste naming.

Does it need to be uppercase?